File: Backend/PixelNestBackend/PixelNestBackend/Dto/CommentDto.cs
using System.ComponentModel.DataAnnotations;

namespace PixelNestBackend.Dto
{
    public class CommentDto
    {
        [Required]
        public string CommentText { get; set; }
        
        [Required]
        public Guid PostID { get; set; }
       
        public int? ParentCommentID { get; set; }
      
    }
}

File: Frontend/PixelNestFrontend/src/app/core/dto/comment.dto.ts
import { LikedByUsers } from "./likedByUsers.dto";
import { Replies } from "./replies.dto";


export interface CommentDto {
    commentID?: number;
    commentText?: string;
    likedByUsers?: LikedByUsers[]; 
    parentCommentID?: number | null;
    postID?: string;
    replies?: Replies[] | [];
    totalLikes?: number;
    clientGuid:string
    username: string;
    totalReplies?:number;
}
  
